index.html

```diff
@@ -929,31 +1106,423 @@ <h2>ConsumedThing Examples</h2>
         } catch(e) {
           console.log("TD fetch error: " + e.message); },
         };
+
         try {
           // subscribe to property change for “temperature”
           await thing.observeProperty("temperature", value => {
-            console.log("Temperature changed to: " + value);
+            console.log("Temperature changed to: " + parseData(value));
           });
           // subscribe to the “ready” event defined in the TD
           await thing.subscribeEvent("ready", eventData => {
-            console.log("Ready; index: " + eventData);
+            console.log("Ready; index: " + parseData(eventData));
             // run the “startMeasurement” action defined by TD
             await thing.invokeAction("startMeasurement", { units: "Celsius" });
             console.log("Measurement started.");
           });
         } catch(e) {
           console.log("Error starting measurement.");
         }
+
         setTimeout( () => {
-          console.log(“Temperature: “ + await thing.readProperty(“temperature”));
+          console.log(“Temperature: “ +
+            parseData(await thing.readProperty(“temperature”)));
           await thing.unsubscribe(“ready”);
           console.log("Unsubscribed from the ‘ready’ event.");
         },
         10000);
+
+        function parseData(response) {
+          if (response.value !== undefined)
+            return data.value;
+          // if response.value is undefined, it means parsing has failed
+          // app can try low-level stream read
+          const reader = value.data.getReader();
+          let value = null;
+
+          reader.read().then(function process({ done, chunk }) {
+              if (done) {
+                value += chunk;
+                return value
+              }
+              value += chunk;
+              return reader.read().then(process);
+            });
+        };
```

Comment:
As in my last comment, is this updated with the latest changes?

Reply:
No, I forgot to update the example. Fixing...

Reply:
Fixed